packages/adapter/addon/rest.js

```diff
@@ -1084,6 +1084,17 @@ const RESTAdapter = Adapter.extend(BuildURLMixin, {
 
     let headers = get(this, 'headers');
     if (headers !== undefined) {
+      if (DEBUG) {
+        let optionsHeaders = Object.keys(options.headers || {});
+        let conflictingHeaders = Object.keys(headers).filter(key => optionsHeaders.includes(key));
+        if (conflictingHeaders.length > 0) {
+          conflictingHeaders.forEach(header => {
+            warn(
+              `${header} defined in adapter will be overridden by ${header} defined in ajaxOptions argument options in the next version of Ember Data.`
```

Comment:
Two things:

1) I think if the values match we shouldn't warn.
2) I think we should print the values.

So something like: `if (optionsHeaders[header] !== conflictingHeaders[header]) { ... }`

```suggestion
              `The request header \`${header}\` with the value \`${optionsHeaders[header]}\` defined on the options passed to \`adapter.ajaxOptions(url, method, options)\` will be overridden with the value \`${conflictingHeaders[header]}\` defined via \`adapter.headers\`.  This behavior will change in the next version of EmberData and the value provided on the \`options\` passed to \`adapter.ajaxOptions\` will be respected.
```